akka-distributed-data/src/main/scala/akka/cluster/ddata/protobuf/ReplicatorMessageSerializer.scala

```diff
@@ -345,7 +345,9 @@ class ReplicatorMessageSerializer(val system: ExtendedActorSystem)
     val get = dm.Get.parseFrom(bytes)
     val key = otherMessageFromProto(get.getKey).asInstanceOf[KeyR]
     val request = if (get.hasRequest()) Some(otherMessageFromProto(get.getRequest)) else None
-    val timeout = Duration(get.getTimeout, TimeUnit.MILLISECONDS)
+    // This incorrectly produces a negative Duration for timeouts larger than
+    // Int.MaxValue, but we don't expect timeouts longer than 24 days anyway.
+    val timeout = Duration(get.getTimeout.toLong, TimeUnit.MILLISECONDS)
```

Comment:
Should it fail if it is negative?

Reply:
Fixed it - seemed easier than explaining the situation :)